cmd/mattermost/commands/config.go

```diff
@@ -343,6 +352,112 @@ func UpdateMap(configMap map[string]interface{}, configSettings []string, newVal
 	}
 }
 
+func configResetCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+
+	defer app.Shutdown()
+
+	oldConfig := app.Config()
+	newConfig := app.Config()
+	defaultConfig := &model.Config{}
+	defaultConfig.SetDefaults()
+
+	if len(args) == 0 {
+		var confirmResetAll string
+		CommandPrettyPrintln("Are you sure you want to reset all the configuration settings?(YES/NO): ")
+		fmt.Scanln(&confirmResetAll)
+		if confirmResetAll == "YES" {
+			oldConfigMap := configToMap(*oldConfig)
+			for key := range oldConfigMap {
+				f := resetConfigValue(key, defaultConfig, oldConfig, newConfig)
```

Comment:
I think it is a bit complex because the "newConfig" pointer used to return a value, i would prefer something like, 

```suggestion
				f := resetConfigValue(key, defaultConfig, currentConfig)
				app.UpdateConfig(f)
				newConfig := app.Config()
```

Of course this means change the other variables in the context, like renaming oldConfig and removing newConfig.